File: dotnet/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/Handlers/GetOrderByIdQueryHandler.cs
using MediatR;
using MediatorDesignPattern.Example2.Models;
using MediatorDesignPattern.Example2.Queries;
using MediatorDesignPattern.Example2.Repositories;

namespace MediatorDesignPattern.Example2.Handlers;

/// <summary>
/// Query handler that retrieves an order by ID
/// </summary>
public class GetOrderByIdQueryHandler : IRequestHandler<GetOrderByIdQuery, Order?>
{
    private readonly IOrderRepository _orderRepository;

    public GetOrderByIdQueryHandler(IOrderRepository orderRepository)
    {
        _orderRepository = orderRepository;
    }

    public Task<Order?> Handle(GetOrderByIdQuery request, CancellationToken cancellationToken)
    {
        var order = _orderRepository.GetById(request.OrderId);
        
        Console.WriteLine(order != null 
            ? $"[HANDLER] Order found: {order.Id}" 
            : $"[HANDLER] Order not found: {request.OrderId}");
        
        return Task.FromResult(order);
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/src/Handlers/GetOrderByIdQueryHandler.ts
import { requestHandler, RequestHandler } from 'mediatr-ts';
import { GetOrderByIdQuery } from '../Queries/GetOrderByIdQuery';
import { Order } from '../Models/Order';
import { orderRepository } from './CreateOrderCommandHandler';

@requestHandler(GetOrderByIdQuery)
export class GetOrderByIdQueryHandler implements RequestHandler<GetOrderByIdQuery, Order | undefined> {
  async handle(request: GetOrderByIdQuery): Promise<Order | undefined> {
    const order = orderRepository.getById(request.orderId);
    console.log(order
      ? `[HANDLER] Order found: ${order.id}`
      : `[HANDLER] Order not found: ${request.orderId}`);
    return order;
  }
}
